cmake/FindFreeimage.cmake

```diff
@@ -9,7 +9,13 @@ if (PKG_CONFIG_FOUND)
     message (STATUS "  freeimage.pc not found, trying freeimage_include_dir and freeimage_library_dir flags.")
   endif (NOT freeimage_FOUND)
 endif (PKG_CONFIG_FOUND)
-
+if (NOT freeimage_FOUND)
+  find_package(freeimage CONFIG)
+  if (freeimage_FOUND)
+    set(freeimage_LIBRARIES freeimage::FreeImage)
+    message (STATUS "FreeImage found")
+  endif()
+endif()
```

Comment:
As for tinyxml, this `find_package(freeimage CONFIG)` is not working, but freeimage is found by the fallback logic:
~~~
2023-08-16T15:47:12.6019654Z CMake Warning at vcpkg/scripts/buildsystems/vcpkg.cmake:855 (_find_package):
2023-08-16T15:47:12.6020579Z   Found package configuration file:
2023-08-16T15:47:12.6021019Z 
2023-08-16T15:47:12.6021899Z     D:/a/gazebo-classic/gazebo-classic/vcpkg/installed/x86-windows-release/share/freeimage/freeimage-config.cmake
2023-08-16T15:47:12.6022448Z 
2023-08-16T15:47:12.6022805Z   but it set freeimage_FOUND to FALSE so package "freeimage" is considered to
2023-08-16T15:47:12.6023299Z   be NOT FOUND.
2023-08-16T15:47:12.6023713Z Call Stack (most recent call first):
2023-08-16T15:47:12.6024195Z   cmake/FindFreeimage.cmake:13 (find_package)
2023-08-16T15:47:12.6024671Z   cmake/SearchForStuff.cmake:6 (include)
2023-08-16T15:47:12.6025135Z   CMakeLists.txt:163 (include)
2023-08-16T15:47:12.6025452Z 
2023-08-16T15:47:12.6025572Z 
2023-08-16T15:47:12.6060283Z --   Found D:/a/gazebo-classic/gazebo-classic/vcpkg/installed/x86-windows-release/include/FreeImage.h
2023-08-16T15:47:14.9965206Z --   Looking for FreeImage.h - found
2023-08-16T15:47:15.0005790Z --   Looking for libfreeimage - found
~~~

There is any specific reason for which we can't just rely on the existing fallback logic?

Reply:
> I think it better to add it correctly if we want later to compile it with vpckg with linux. I will try to unset the the same.

We can do changes required for other PRs in the corresponding PR. In general, it is ideal for facilitate review that the PR as as small as possible, without unrelated changes, see https://gazebosim.org/docs/all/contributing and https://google.github.io/eng-practices/review/developer/small-cls.html .

Reply:
If you insist. I will revert these changes.

Reply:
I revert the changes.